docs/DeveloperGuide.adoc

```diff
@@ -154,89 +145,86 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Given below is the example usage of how the the analyse functionality, using the `DictionaryAnalyser`, behaves at each step.
 
-image::UndoRedoState1.png[]
+Step 1. The user inputs `analyse` into the command line. The `Logic` uses the `PasswordBookParser` to parse the user command,
+which results in a `AnalysePasswordCommand` which is executed by the `LogicManager`.
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+Step 2. The `AnalysePasswordCommand` calls the `Model#getFilteredPasswordList()` to obtain the current list of passwords,
+as well as `#getRequiredAnalysers()` to get the default list of analysers.
 
-image::UndoRedoState2.png[]
+Step 3. `AnalysePasswordCommand` will invoke `Analyser#analyse()` of each analyser on every password in the list to produce the respective result.
+The `Result` object contains information of the password after analysis, such as the description of result.
+For comparison-based analysers, `Match` objects are created to store information about matches found by the analyser.
+Each `Match` contains details such as the match token, as well as the start and end index of the token with respect to the password.
 
-[NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+Step 4. A report is compiled by stringify-ing the list of `Result` objects from each analyser and shown to the user.
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+Step 5. Should the user wish to view a more detailed report for a specific password, user may input `analyse strong/<index>`
+to view a more detailed analysis of the specified password.
 
-image::UndoRedoState3.png[]
+The following sequence diagram shows how the analyse functionality works:
 
-[NOTE]
-If the `currentStatePointer` is at index 0, pointing to the initial address book state, then there are no previous address book states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the undo.
+image::AnalysePasswordSequenceDiagram.png[]
```

Comment:
The activation bar for DictionaryAnalyser self-called method getAllMatches() seems to be missing a return arrow. Also, the DictionaryMatch return arrow would require the return arrow head to be pointing within the activation bar instead of at the end of the bar.
![image](https://user-images.githubusercontent.com/26473264/67543437-f225f280-f723-11e9-81b6-def2461ff3ac.png)